### ISSUE TITLE
Expose challenge details via API

### ISSUE THREAD

**Issue by jmboyd** [magwo/elevatorsaga#66]
Love to have API access to a **challenge** or **world** object that would allow adaptive algorithms to know the specs of the current challenge -- in particular, challenge type, challenge parameters, spawn rate, and elevator sizes (though the elevator size would probably fit better in the **elevator** object.)

Currently we have to resort to counting floors and elevators or scraping the challenge number from the URL, but that's not too pretty... plus needs adjustment with each change to challenge.js.

**Comment by magwo** [comment]
Interesting request. I'm not sure how much should be exposed.

Elevator sizes are obvious, it just hasn't been done yet. Others I will have to think about. I don't want players to have to optimize their algorithms for specific challenges or challenge types. The game should be about developing one algorithm, not specific hacks or optimizations to pass specific challenges. At least that's my view. Hmm.

**Comment by jmboyd** [comment]
That's my ideal goal as well, but I've found the **requireUserCountWithinMoves** challenges in particular need significantly different behavior for a decent success rate. Judging from the posted solutions, other solvers have found likewise: Solutions have constants that need to be adjusted for the **requireUserCountWithinMoves** challenges, or have two entirely separate code sets, or attempt to infer from the challenge number scraped from URL or floor/elevator counts (assuming a relatively stable challenges.js), or simply choke on numbers 6 and 7.

I see we do get **maxPassengerCount** now though -- thanks for that one!

**Comment by magwo** [comment]
It has become obvious to me that the requireUserCountWithinMoves challenge is a bad idea. It should be replaced with a challenge that can not be "hacked". Probably something like requireUserCountWithinMovesAndTime.

Will probably not expose challenge details to player code. What COULD be exposed, is perhaps a general challenge optimisation focus, or a list of them, like for example: challengeType: ["time", "moveCount"].

**Comment by Rybadour** [comment]
> It has become obvious to me that the requireUserCountWithinMoves challenge is a bad idea.

Agreed, when I first ran into this type I gutted 99% of my code, rewrote it at 10% of the size, and preceded to complete the next 3 challenges of the same type. Because there is no time limit you only have to completely fill an elevator then move.
